File: Day 6/MajorityElement.cpp
#include<bits/stdc++.h>
using namespace std;


class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> umap;
        for(int i=0; i<nums.size(); i++){
            umap[nums[i]]++;
        }

        for(int i=0; i<nums.size(); i++){
            if(umap[nums[i]] > nums.size()/2){
                return nums[i];
            }
        }
        return -1;
    }
};

File: Day 6/MajorityElement.java
import java.util.HashMap;

class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> umap = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            umap.put(nums[i], umap.getOrDefault(nums[i], 0)+1);
        }

        for(int i=0; i<nums.length; i++){
            if(umap.get(nums[i]) > (nums.length/2)){
                return nums[i];
            }
        }
        return -1;
    }
}